File: KtvAssistant_ios/SinaKTVAide/Model/DataStruct/RoomInfo.m
//
//  RoomInfo.m
//  SinaKTVAide
//
//  Created by apple on 13-10-29.
//  Copyright (c) 2013年 Zhang Chen. All rights reserved.
//

#import "RoomInfo.h"
#import "PCommonUtil.h"

@implementation RoomInfo

@synthesize ktvId = _ktvId;
@synthesize roomId =  _roomId;
@synthesize roomName = _roomName;
@synthesize beginTime = _beginTime;
@synthesize loginTime = _loginTime;
@synthesize periodsTime = _periodsTime;
@synthesize address = _address;
@synthesize roomPassword = _roomPassword;
@synthesize xgRoomID = _xgRoomID;

+(RoomInfo *)initWithDefaultValue{
    
    RoomInfo *roomInfo = [[RoomInfo alloc] init];
    roomInfo.ktvId = @"0";
    roomInfo.roomId = @"0";
    roomInfo.roomPassword = @"";
    roomInfo.xgRoomID = @"0";
    return roomInfo;
}

+(RoomInfo *)initWithDictionary:(NSDictionary *)data{
    
    RoomInfo *roomInfo = nil;
    
    @try {
        if (data && [data isKindOfClass:[NSDictionary class]]) {
            roomInfo = [[RoomInfo alloc] init];
            roomInfo.ktvId = [data objectForKey:@"ktvid"];
            roomInfo.roomId = [NSString stringWithFormat:@"%@",[data objectForKey:@"roomid"]];
            roomInfo.roomName = [PCommonUtil decodeBase64:[data objectForKey:@"roomname"]];
            roomInfo.beginTime = [PCommonUtil decodeBase64:[data objectForKey:@"begintime"]];
            roomInfo.loginTime = [PCommonUtil decodeBase64:[data objectForKey:@"logintime"]];
            roomInfo.periodsTime = [[data objectForKey:@"periodstime"] longLongValue];
            roomInfo.address = [PCommonUtil decodeBase64:[data objectForKey:@"address"]];
            roomInfo.xgRoomID = [data objectForKey:@"xgroomid"];
        }
    }
    @catch (NSException *exception) {
        PLog(@"create RoomInfo failed...please check");
    }
    
    return roomInfo;
}

-(void)Log{
    PLog(@"Print RoomInfo: _ktvId(%@), _roomId(%@), _roomPassword(%@), _roomName(%@), _beginTime(%@), _loginTime(%@), _periodsTime(%lld), _address(%@)", _ktvId, _roomId, _roomPassword, _roomName, _beginTime, _loginTime, _periodsTime, _address);
}

@end

File: KtvAssistant_android/src/assistant/entity/RoomInfo.java
package assistant.entity;

import org.json.JSONObject;

import assistant.task.PCommonUtil;
import assistant.util.ShowLog;


public class RoomInfo {
	
	public String ktvId;		//实体店id
	public String roomId;
	public String roomName;
	public String beginTime;	//包厢开启时间
	public String loginTime;	//客户端登录时间
	public long periodsTime;	//单位秒
	public String address;		//实体店地址
	
	public RoomInfo(){
		this.ktvId = "0";
		this.roomId = "0";
	}
	
	public RoomInfo(JSONObject jsonObject){
		
		try {
			if(null != jsonObject){
				this.ktvId = jsonObject.optString("ktvid", "0");
				this.roomId = jsonObject.optString("roomid");
				this.roomName = PCommonUtil.decodeBase64(jsonObject.optString("roomname"));
				this.beginTime = PCommonUtil.decodeBase64(jsonObject.optString("begintime"));
				this.loginTime = PCommonUtil.decodeBase64(jsonObject.optString("logintime"));
				this.periodsTime = jsonObject.optLong("periodstime");
				this.address = PCommonUtil.decodeBase64(jsonObject.optString("address"));
			}
			
		} catch (Exception e) {
			ShowLog.i("Parser RoomInfo failed...please check");
		}
		
	}
	
	public void log() {
		//ShowLog.i("Print RoomInfo: ktvId(" + ktvId + "), roomId(" + roomId + "), roomName(" + roomName + "), beginTime(" + beginTime + "), loginTime(" + loginTime + "), periodsTime(" + periodsTime + "), address(" + address + ")");
	}
	
}
